### ISSUE TITLE
updateOptions Crashing in FirestoreRecyclerAdapter

### ISSUE THREAD

**Issue by abhilashi** [firebase/FirebaseUI-Android#1707]
The following commit added ##updateOptions##

https://github.com/firebase/FirebaseUI-Android/commit/d7aec385c2bbb807b39b4c3c444f161f55be17fc

https://github.com/firebase/FirebaseUI-Android/commit/d7aec385c2bbb807b39b4c3c444f161f55be17fc#r35785637

But it's half done. For example, the ##FirestoreRecyclerAdapter## constructor isn't updated and tested.

https://github.com/firebase/FirebaseUI-Android/commit/d7aec385c2bbb807b39b4c3c444f161f55be17fc#diff-5e8b0d86c8d693eb2093b88e08e1e76f

#### Steps to reproduce:

import com.firebase.ui.firestore.FirestoreRecyclerAdapter
and updateOptions on it
  
#### Observed Results:

  * What happened?  This could be a description, `logcat` output, etc.
  * NullPointerException on line 105: https://github.com/firebase/FirebaseUI-Android/commit/d7aec385c2bbb807b39b4c3c444f161f55be17fc#r35785637
  
#### Expected Results:

  * What did you expect to happen?
  * Options being actually updated
  
#### Relevant Code:

  ```
messagesInChatAdapter.updateOptions(messagesInChatOptions as FirestoreRecyclerOptions<MessageModel>)
  ```

**Comment by abhilashi** [comment]
@samtstern  @PatilShreyas Could you pl take a look at this?

**Comment by PatilShreyas** [comment]
@abhilashi Still can't understand What you want to do. Could you provide code?

**Comment by abhilashi** [comment]
@PatilShreyas thanks much for the quick reply. It's a simple issue. 

There are two ##FirestoreRecyclerAdapters## not one. One in ##ui/firestore## directory and other in ##ui/database##

You added ##updateOptions## in both of them but forgot to change the constructor in one of them. In ##ui/database/FirestoreRecyclerAdapter## _mOptions_ is updated and it works fine. 

But the same isn't done in ##ui/firestore/FirestoreRecyclerAdapter##. See lines 38 and 39  https://github.com/firebase/FirebaseUI-Android/blob/d7aec385c2bbb807b39b4c3c444f161f55be17fc/firestore/src/main/java/com/firebase/ui/firestore/FirestoreRecyclerAdapter.java#L38

@PatilShreyas @samtstern Let me know if you need more details. I can give the code but this is more succinct way to get to the root of the issue.

**Comment by PatilShreyas** [comment]
First of all, there are different adapters
- `FirebaseRecyclerAdapter`
- `FirestoreRecyclerAdapter`

This API is tested before merging into repo. Can you provide your
implemention so that I can assist you further.

**Comment by PatilShreyas** [comment]
@abhilashi First of all, there are different adapters
`FirebaseRecyclerAdapter` and 
`FirestoreRecyclerAdapter`

This API is tested before merging into repo. Can you provide your implemention so that I can assist you further.

**Comment by abhilashi** [comment]
@PatilShreyas thanks again for quick response. 

The problem is in _FirestoreRecyclerAdapter_ 

Request you to look at your commit history on this adapter: _FirestoreRecyclerAdapter_ 

The very first time _updateOptions_ is called, _mOptions_ is _null_.

**Comment by abhilashi** [comment]
Here is the exception trace: 

```
java.lang.NullPointerException: Attempt to invoke virtual method 'androidx.lifecycle.LifecycleOwner com.firebase.ui.firestore.FirestoreRecyclerOptions.getOwner()' on a null object reference
        at com.firebase.ui.firestore.FirestoreRecyclerAdapter.updateOptions(FirestoreRecyclerAdapter.java:106)
        at com.feelr.MainActivity.initOrUpdateMessagesInChat(MainActivity.kt:378)
        at com.feelr.MainActivity.onChatChanged(MainActivity.kt:114)
        at com.feelr.MainActivity.access$onChatChanged(MainActivity.kt:52)
        at com.feelr.MainActivity$initOrUpdateFriendsInNavDrawer$1$onBindViewHolder$1.onClick(MainActivity.kt:407)
        at android.view.View.performClick(View.java:6597)
        at android.view.View.performClickInternal(View.java:6574)
        at android.view.View.access$3100(View.java:778)
        at android.view.View$PerformClick.run(View.java:25889)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6746)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

**Comment by abhilashi** [comment]
@PatilShreyas @samtstern pl let me know if you need more details.

**Comment by PatilShreyas** [comment]
@abhilashi Please provide full activity implementation to trace what is going on.

**Comment by abhilashi** [comment]
@PatilShreyas the code is very simple - just calls updateOptions on FirestoreRecylerAdapter. If you want me to provide a sample activity I'll need to write one to give you. But if you have already tested before committing the code, could you please re-run those tests?

**Comment by abhilashi** [comment]
@PatilShreyas @samtstern here is the code snippet that's throwing up. The call to _updateOptions_ below.
> 
>        messagesInChatOptions = FirestoreRecyclerOptions.Builder<MessageModel>()
>             .setQuery(messagesInChatQuery, MessageModel::class.java)
>             .setLifecycleOwner(this)
>             .build()
> 
>         if (messagesInChatAdapter == null) {
>             messagesInChatAdapter = object :
>                 FirestoreRecyclerAdapter<MessageModel, MessageInChatViewHolder>(
>                     messagesInChatOptions!!
>                 ) {
>                 override fun onBindViewHolder(
>                     holder: MessageInChatViewHolder,
>                     position: Int,
>                     model: MessageModel
>                 ) {
>                    <REDACTED>
>                 }
> 
>                 override fun getItemViewType(position: Int): Int {
>                     <REDACTED>
>                 }
> 
>                 override fun onCreateViewHolder(
>                     parent: ViewGroup,
>                     viewType: Int
>                 ): MessageInChatViewHolder {
>                     return MessageInChatViewHolder(view)
>                 }
> 
>                 override fun onDataChanged() {
>                  
>                 }
>             }
>             messagesInChatRecyclerView.adapter = messagesInChatAdapter
>         } else {
>             messagesInChatAdapter?.updateOptions(messagesInChatOptions as FirestoreRecyclerOptions<MessageModel>)
>         }

**Comment by PatilShreyas** [comment]
@abhilashi Thank you for this. Now I actually came to know what exactly happened. In `FirestoreRecyclerAdapter`, assignment of `options` to `mOptions` is missing in constructor that's why *NullPointerException* is being thrown. 
It's my mistake!

@samtstern This issue should be fixed before releasing the new version.

**Comment by samtstern** [comment]
Noted, thank you!

**Comment by samtstern** [comment]
This has been released in FirebaseUI `6.1.0`
